lib/mix/tasks/phx.gen.html.ex

```diff
@@ -184,7 +184,15 @@ defmodule Mix.Tasks.Phx.Gen.Html do
         ~s(<.input field={{f, #{inspect(key)}}} type="number" label="#{label(key)}" step="any" />)
 
       {key, :boolean} ->
-        ~s(<.input field={{f, #{inspect(key)}}} type="checkbox" label="#{label(key)}" />)
+        ~s"""
+        <.input
+          field={{f, #{inspect(key)}}}
+          type="checkbox"
+          label="#{label(key)}"
+          value="true"
+          checked={to_string(input_value(f, #{inspect(key)})) == "true"}
```

Comment:
I tried moving this logic to the component but it ended up being more complicated than having the value and the condition to check here.

Some of the variations I tested was from https://github.com/phoenixframework/phoenix/pull/5000#discussion_r987467759.